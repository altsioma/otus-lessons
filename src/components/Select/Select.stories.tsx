import React from "react";
import { Select } from "../../components/Select/Select";
import { action } from "@storybook/addon-actions";

import { withKnobs } from "@storybook/addon-knobs";

export default {
  title: "GraphMakerComponents",
  decorators: [withKnobs],
};

const selectChange = action("Select change");

export const SelectControl: React.FC<{}> = () => (
  <Select
    selected="label"
    values={[
      { label: "test label 1", value: "test value" },
      { label: "test numeric", value: 10 },
      { label: "just string", value: "string" },
    ]}
    onChange={(e) => {
      selectChange(e.target.value);
    }}
  />
);
