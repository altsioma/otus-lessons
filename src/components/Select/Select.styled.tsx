import React from "react";
import styled from "@emotion/styled";

export const SelectStyled = styled.select`
  display: inline-block;
  vertical-align: middle;
  border-radius: 4px;
  padding: 11px 5px;
  margin: 5px;
  background-color: white;
  color: black;
  cursor: pointer;
  border: none;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
`;
