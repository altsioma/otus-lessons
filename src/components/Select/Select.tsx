import React from "react";
import styled from "@emotion/styled";

type SelectType = {
  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
  selected: string | number;
  values: { label: string; value: string | number }[];
};

const SelectWrapper = styled.select`
  display: inline-block;
  vertical-align: middle;
  border-radius: 4px;
  padding: 11px 5px;
  margin: 5px;
  background-color: white;
  color: black;
  cursor: pointer;
  border: none;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
`;
export const Select: React.FC<SelectType> = ({
  selected,
  onChange,
  values,
}) => {
  let i = 0;
  return (
    <SelectWrapper defaultValue={selected} onChange={onChange}>
      {values.map((data) => (
        <option key={i++} value={data.value}>{data.label}</option>
      ))}
    </SelectWrapper>
  );
};
