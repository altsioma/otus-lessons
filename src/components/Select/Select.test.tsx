import React from "react";
import { Select } from "./Select";
import renderer from "react-test-renderer";

describe("Select", () => {
  const values: { label: string; value: string }[] = [
    { label: "0.5x", value: "0.5" },
    { label: "Test text", value: "text" },
  ];
  it("renders Select with two options", () => {
    expect(
      renderer
        .create(<Select selected="text" values={values} onChange={jest.fn()} />)
        .toJSON()
    ).toMatchSnapshot();
  });
});
