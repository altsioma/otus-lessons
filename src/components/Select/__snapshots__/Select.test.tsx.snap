// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders Select with two options 1`] = `
<select
  className="css-10x8iy2"
  defaultValue="text"
  onChange={[MockFunction]}
>
  <option
    value="0.5"
  >
    0.5x
  </option>
  <option
    value="text"
  >
    Test text
  </option>
</select>
`;
